src/moi.jl

```diff
@@ -255,6 +255,52 @@ end
 accept_vector_set(::ProductMode{T}, ::Complement) where T = nothing
 accept_vector_set(::MixedMode{T}, ::Complement) where T = nothing
 
+function get_variable_complement(primal_model, dual_model, primal_con, dual_con) 
+    error("An internal error with variable complements occurred. Likely, your problem type does not yet support consideration of constrained variables.")
+end
+
+function get_variable_complement(primal_model, dual_model, primal_con::MOI.ConstraintIndex{Fp,Sp}, dual_con::MOI.ConstraintIndex{Fd,Sd}) where {Fp<:MOI.VariableIndex,Sp<:MOI.GreaterThan{T},Fd,Sd<:MOI.GreaterThan{T}} where T
```

Comment:
any reason for not supporting other sets like vector sets?
you can follow the code in get_canonical_complement, as you have been doing

Reply:
left this out because I did not really feel certain. 
Now added, can you check if it is correct? 
Also not sure about the todo (see code for the new function)